standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -10614,6 +10644,101 @@ public void dropStoredProcedure(String catName, String dbName, String funcName)
     return names;
   }
 
+  @Override
+  public void addPackage(Package request) throws NoSuchObjectException, MetaException {
+    boolean committed = false;
+    MPackage mPkg;
+    Query query = null;
+    String catName = normalizeIdentifier(request.getCatName());
+    String dbName = normalizeIdentifier(request.getDbName());
+    MDatabase db = getMDatabase(catName, dbName);
+    try {
+      openTransaction();
+      query = findPackageQuery();
+      mPkg = (MPackage) query.execute(request.getPackageName(), dbName, catName);
+      pm.retrieve(mPkg);
+      if (mPkg == null) { // create new
+        mPkg = new MPackage();
+        populate(request, db, mPkg);
+        pm.makePersistent(mPkg);
+      } else { // update existing
+        populate(request, db, mPkg);
+      }
```

Comment:
I think this is an add and update 2-in-1 because you have 2 add methods in the `Package` interface